coreneuron/io/reports/report_handler.cpp

```diff
@@ -11,7 +11,7 @@ void ReportHandler::create_report(double dt, double tstop, double delay) {
     }
     m_report_config.stop = std::min(m_report_config.stop, tstop);
 
-    m_report_config.mech_id = nrn_get_mechtype(m_report_config.mech_name);
+    m_report_config.mech_id = nrn_get_mechtype(m_report_config.mech_name.data());
```

Comment:
Is it `.data()` or `.c_str()`? The code might expect a null-terminated string, so this could be dangerous, right?

Reply:
Actually it just converts the char* to string, so .data() would be just fine here!
```
int nrn_get_mechtype(const char* name) {
    std::string str(name);
```
And i cant just pass the string because the function is used in lots other places...

Reply:
Actually, from C++11 are equivalent